jsonrpc2_test.go

```diff
@@ -314,105 +314,88 @@ type noopHandler struct{}
 
 func (noopHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrpc2.Request) {}
 
-type readWriteCloser struct {
-	read, write func(p []byte) (n int, err error)
-}
-
-func (x readWriteCloser) Read(p []byte) (n int, err error) {
-	return x.read(p)
-}
-
-func (x readWriteCloser) Write(p []byte) (n int, err error) {
-	return x.write(p)
-}
+func TestConn_DisconnectNotify(t *testing.T) {
 
-func (readWriteCloser) Close() error { return nil }
+	t.Run("EOF", func(t *testing.T) {
+		connA, connB := net.Pipe()
+		c := jsonrpc2.NewConn(context.Background(), jsonrpc2.NewPlainObjectStream(connB), nil)
```

Comment:
I switched from `jsonrpc2.NewBufferedStream(conn, jsonrpc2.VarintObjectCodec{}), nil)` to `jsonrpc2.NewPlainObjectStream(conn)` solely because the latter looks cleaner/shorter. The type of stream being used shouldn't impact the results of the these tests. I am happy to switching back to the former though if you prefer.